src/ng/directive/validators.js

```diff
@@ -68,15 +68,21 @@ var requiredDirective = ['$parse', function($parse) {
     require: '?ngModel',
     link: function(scope, elm, attr, ctrl) {
       if (!ctrl) return;
-      var value = attr.required || $parse(attr.ngRequired)(scope);
+      // For boolean attributes like required, presence means true
+      var value = 'required' in attr || $parse(attr.ngRequired)(scope);
 
-      attr.required = true; // force truthy in case we are on non input element
+      if (!attr.ngRequired) {
+        // force truthy in case we are on non input element
+        // (input elements do this automatically for boolean attributes like required)
+        attr.required = true;
```

Comment:
Shouldn't this be consistent with `value` above?

Reply:
What do you mean? If ngRequired isn't set, we must be in the required path. This condition is just for non-input elements.

Reply:
NVM, I forgot that at this point either `attr.required` or `attr.ngRequired` are defined 👍

Reply:
I forgot this about two times myself while working on this 😄